src/lib/OpenEXR/ImfCompositeDeepScanLine.cpp

```diff
@@ -427,6 +427,23 @@ LineCompositeTask::execute ()
 
 } // namespace
 
+
+namespace {
+int64_t maximumSampleCount = 1<<25;
```

Comment:
should we prefer the same behaviour as for the max image / tile size and have this default to 0 to indicate no limit and in the fuzz test initialize it to 64 or something to keep it low there since they run on tiny machines the way we do w/ the image size?

Reply:
I'm happy to change the logic if that's the consensus. I had considered the alternative: that the limits are 'off by default' and enabled if reduceMemory or reduceTime are enabled in CheckFile().

My thinking for making it 'on by default' was that it may not be widely known that deep images are supported by the InputFile API, and that certain files would take much more time and memory to read than OpenEXR users might expect. By forcing the change on, we protect code that was never intended to support deep images, and the limit is high enough that it's unlikely to inconvenience any workflows. (If you generally have more than 1<<25 samples on a scanline you probably should read the file using the deep API, and probably should have written the file using small tiles instead of scanlines)

As you say, the disadvantage of the 'on by default' is that it is inconsistent both with the existing image and tile size limits, and with older versions of OpenEXR.